packages/example/src/containers/Dashboard/Dashboard.tsx

```diff
@@ -42,50 +31,25 @@ export const Dashboard = () => {
         setNetwork(networkName);
       };
 
-
-    const testFunc = async (): Promise<IPolkadotAdapter | void> => {
-        const allInjected = await web3Enable('my cool dapp');
-        let MetamaskPolkadotSnap: IPolkadotAdapter;
-
-        allInjected.forEach(extension => {
-            console.log(extension.name);
-            if(extension.name==="metamask-polkadot-snap") {
-                console.log("ddaad")
-                MetamaskPolkadotSnap = extension as IPolkadotAdapter;
-                return MetamaskPolkadotSnap;
-            } else console.log( "no polkadot adapter");
-        });
-        // const MetamaskPolkadotSnap = allInjected as unknown as IPolkadotAdapter;
-        // MetamaskPolkadotSnap.getMetamaskSnapApi
-
-        
-        // return allInjected as unknown as IPolkadotAdapter;
+    const getMetamask = (item: InjectedExtension): boolean => {
+        if(item.name==="metamask-polkadot-snap") return true;
+        else return false
     }
 
-    injectMetamaskPolkadotSnapProvider(
-        "kusama"
-        );
     useEffect(() => {
         (async () => {
-            const allInjected = await web3Enable('my cool dapp');
-            console.log(allInjected);
-            const test = await testFunc();
-            console.log(test);
-            // console.log(allInjected[0].g);
-            // allInjected["metamask-js"].getMetamaskApi().configure()
-            // const allAccounts = await web3Accounts();
-            // console.log(allAccounts);
-            // const injector = await web3FromAddress('EHkbSaapJPXsL1F83jpw6dWcN8Ar8PD1JDyKoTZVKCDF4J7');
-            // console.log(injector);
-            // const allProviders = web3ListRpcProviders('metamask-polkadot-snap');
-            // console.log(allProviders);
+            
             if(state.polkadotSnap.isInstalled) {
-                // await setConfiguration({networkName: network});
-                // await addPolkadotAsset();
-                // setPublicKey(await getPublicKey());
-                // setAddress(await getAddress());
-                // setBalance(await getBalance());
-                // setLatestBlock(await getLatestBlock());
+                const allInjected = await web3Enable('my cool dapp');
+                const provider = allInjected.find(getMetamask) as unknown as InjectedMetamaskExtension;
```

Comment:
usually I wouldn't extract getMetamask to separate function, keep it inline as anonymus function:
```javascript
allInjected.find((name => name==="metamask...")
```
also check if provider is not null.